include/robot_interfaces/sensors/sensor_logger.hpp

```diff
@@ -145,7 +145,13 @@ class SensorLogger
     //! Get observations from sensor_data_ and add them to the buffer.
     void loop()
     {
-        auto t = sensor_data_->observation->oldest_timeindex();
+        // Get the oldest available timeindex as starting point of the log.  In
+        // case there is no data yet (t == EMPTY), start with t = 0.
+        auto t = sensor_data_->observation->oldest_timeindex(false);
```

Comment:
why using 'auto' for t ? Why not time_series::Index ?

Reply:
I think this still dates back to the time series from real_time_tools, where the index type was a bit more complex.  I agree that it's better to be explicit here.